src/Stratis.Bitcoin.Features.SmartContracts/Rules/AllowedScriptTypeRule.cs

```diff
@@ -60,11 +62,19 @@ private void CheckOutput(TxOut output)
             if (PayToPubkeyHashTemplate.Instance.CheckScriptPubKey(output.ScriptPubKey))
                 return;
 
+            // For cross-chain transfers
+            if (PayToScriptHashTemplate.Instance.CheckScriptPubKey(output.ScriptPubKey))
+                return;
+
             // For cross-chain transfers
             if (PayToMultiSigTemplate.Instance.CheckScriptPubKey(output.ScriptPubKey))
                 return;
 
-            new ConsensusError("disallowed-output-script", "Only P2PKH and smart contract scripts are allowed.").Throw();
+            // For cross-chain transfers
+            if (TxNullDataTemplate.Instance.CheckScriptPubKey(output.ScriptPubKey))
```

Comment:
Is the size here not a problem?

I guess not, this are rules for the POA network right?
So effectively we allow opreturn of size of 83 bytes *similar to BTC).